business_rules/operators.py

```diff
@@ -1014,6 +1014,35 @@ def value_has_multiple_references(self, other_value: dict) -> pd.Series:
     def value_does_not_have_multiple_references(self, other_value: dict) -> pd.Series:
         return ~self.value_has_multiple_references(other_value)
 
+    @type_operator(FIELD_DATAFRAME)
+    def target_is_sorted_by(self, other_value: dict) -> pd.Series:
+        """
+         Checking if  SESEQ is  sorted based on SESTDTC within USUBJID
+        """
+        target: str = self.replace_prefix(other_value.get("target"))
+        within: str = self.replace_prefix(other_value.get("within"))
+        columns = other_value["comparator"]
+
+        for col in columns:
+            comparator: str = self.replace_prefix(col["name"])
+            sort_as: str = col["sort_order"]
+            na_pos: str = col["null_position"]
+
+            ascending = sort_as != "DESC"
+            grouped_df = self.value.sort_values(
+                by=[within, comparator],
+                na_position=na_pos
+            ).groupby([within]).apply(lambda x: x)
```

Comment:
Why do we need `.apply(lambda x: x)` here?

Reply:
To convert the DataFrameGroupBy object to a Series